fake-toolchain/Unix/cc

```diff
@@ -9,6 +9,6 @@ elif [ "$1" = "-v" ] || [ "$1" = "--version" ]; then
     $YCM_CONFIG_GEN_CC_PASSTHROUGH $@
 
 else
-    echo "$@" >> $YCM_CONFIG_GEN_CC_LOG
+	echo "YCM_CONFIG_GEN_PWD=$(pwd) $@" >> $YCM_CONFIG_GEN_CC_LOG
```

Comment:
Ah I see.  While that will work for `YCM_CONFIG_GEN_PWD`, it seems that spaces in paths coming from the compiler invocation will still parse incorrectly since the quotes are dropped by `$@`.  I eagerly await your explanation of what I've missed this time ;)